chains/evm/listener/listener.go

```diff
@@ -99,8 +103,9 @@ func (l *EVMListener) ListenToEvents(ctx context.Context, startBlock *big.Int, m
 				err := handler.HandleEvent(startBlock, new(big.Int).Sub(endBlock, big.NewInt(1)), msgChan)
 				if err != nil {
 					l.log.Error().Err(err).Msgf("Unable to handle events")
-					continue
+					continue loop
```

Comment:
But won't there be a problem if there is an error not on the first iteration but rather on some second handler after for example some deposits already handled, this will mean that those deposits will be handled one more time?

Reply:
We expect that already can happen anyway for transfers (except for refresh and keygen processes which are possible to run multiple times, but I think those stop everything in a block anyway).

In general, event handlers have to be written so they can be reproducable or else we wouldn't be able to retry any block which is a much more problematic situation.